monix-reactive/shared/src/main/scala/monix/reactive/observables/ObservableLike.scala

```diff
@@ -2183,6 +2183,14 @@ trait ObservableLike[+A, Self[+T] <: ObservableLike[T, Self]]
   /** Zips the emitted elements of the source with their indices. */
   def zipWithIndex: Self[(A, Long)] =
     self.liftByOperator(new ZipWithIndexOperator[A])
+
+  /** Creates a new observable from this observable that will emit a specific `separator`
+    * between every pair of elements.
+    *
+    * @param separator is the separator
+    */
+  def intersperse[B >: A](separator: B): Self[B] =
+    self.transform(self => new IntersperseObservable(self, separator))
```

Comment:
If we introduce this, we also need the version with the `start` and the `end` tokens and I think it can be handled by the same implementation - in that `Observable` source you can make them optional or something.